File: POO/js/persona.js
class Persona {
<<<<<<< Updated upstream
    constructor(nombre, edad, carrera,universidad) {
=======
    constructor(nombre, edad, carrera, universidad) {
>>>>>>> Stashed changes
      this.nombre = nombre;
      this.edad = edad;
      this.carrera = carrera;
      this.universidad = universidad;
    }
  
    //CREAR
    saludar() {
      return 'Hola, mi nombre es ' + this.nombre;
    }
  
    cumpleanios() {
      return ' Tengo ' + this.edad + ' años.';
    }
  
    estudiar() {
      return 'Estoy estudiando ' + this.carrera;
    }

    uni() {
<<<<<<< Updated upstream
      return 'Estudio en la' + this.universidad;
    }
  
  
    //MODIFICAR
    modificar_saludo() {
      this.nombre = prompt('Ingrese un nuevo nombre');
        while (/\d/.test(this.nombre) || this.nombre.trim() === '') {
          alert('El nombre no debe contener digitos o estar vacio la entrada. Inténtelo de nuevo.');
          this.nombre = prompt('Ingrese un nuevo nombre');
      }
      return '¡Hola, mi nombre es ' + this.nombre;
=======
        return 'Me encuentro en la' + this.universidad;
    }
>>>>>>> Stashed changes
  }

    modificar_cumpleanios() {
      this.edad = prompt('Ingrese su edad nuevamente');
    while (!/^\d+$/.test(this.edad) || parseInt(this.nombre)<= 0) {
        alert("La edad no puede ser un número negativo y estar vacio la entrada. Intentalo de nuevo.");
        this.edad = prompt('Ingrese su edad');
      }
      return 'Tengo ' + this.edad + ' años.';
    }

    modificar_estudiar() {
      this.carrera = prompt('Ingrese tu carrera');
      while (/\d/.test(this.carrera) || this.carrera.trim() === '') {
        alert('No debe estar vacio la entrada, ni contener digitos. Intentalo de nuevo.');
        this.carrera = prompt('Ingrese tu carrera');
      }
      return 'Estoy estudiando '  + this.carrera;
    }

    modificar_universidad() {
      this.universidad = prompt("Ingrese el nombre de tu universidad")
      while (/\d/.test(this.universidad) || this.universidad.trim() === '') {
        alert('No debe estar vacio la entrada, ni contener digitos. Intentalo de nuevo.');
        this.universidad = prompt('Ingrese el nombre de tu universidad ');
      }
      return'En la  ' +  this.universidad;
    }

  
    //ELIMINAR

    eliminar_nombre() {
      this.nombre = undefined;
      
    }

    eliminar_edad() {
      this.edad = undefined;
    }

    eliminar_carrera() {
      this.carrera = undefined;
    }

    eliminar_universidad() {
      this.universidad = undefined;
    }

  }




  // Crear una instancia de la clase Persona
<<<<<<< Updated upstream
  const persona = new Persona('Miguel', 18, 'Ing. Industrial', ' UCB');
=======
  const persona = new Persona( 'Miguel', 18, 'Ingeniería Industrial',' U. Catolica Boliviana');
  
 
  
  

>>>>>>> Stashed changes
  

  // Inicializar los elementos HTML
  const saludar = document.getElementById('saludar');
  const edad = document.getElementById('edad');
  const carrera = document.getElementById('carrera');
<<<<<<< Updated upstream
  const uni = document.getElementById('universidad');


=======
  const Universidad = document.getElementById('Universidad');
>>>>>>> Stashed changes
  
 
  
  // Función para saludar
  function botonSaludar() {
    saludar.textContent = persona.saludar();
  }
  
  // Función para cumpleaños
  function botonCumpleanios() {
    edad.textContent = persona.cumpleanios();
  }
  
  // Función para estudiar
  function botonEstudiar() {
    carrera.textContent = persona.estudiar();
  }

<<<<<<< Updated upstream
  // Funcion para universidad
  function botonUni() {
    uni.textContent = persona.uni();
  }

  // funcion para modificar_saludo
  function botonModificar_Saludar() {
    saludar.textContent = persona.modificar_saludo();
  }

  //funcion para modificar_cumpleanos
  function botonModificar_Cumple() {
    edad.textContent = persona.modificar_cumpleanios();
  }

  //funcion para modificar_carrera
  function botonModificar_Carrera() {
    carrera.textContent = persona.modificar_estudiar();
  }

  //function para modificar_universidad
  function botonModificar_Universidad() {
    uni.textContent = persona.modificar_universidad();
  }



  //Eliminar
  function botonEliminar_Nombre() {
    saludar.textContent = persona.eliminar_nombre();
  }

  function botonEliminar_Edad() {
    edad.textContent = persona.eliminar_edad();
  }

  function botonEliminar_Carrera() {
    carrera.textContent = persona.eliminar_carrera();
  }

  function botonEliminar_Universidad() {
    uni.textContent = persona.eliminar_universidad();
=======
  //Funcion para universidad
  function botonUni() {
    Universidad.textContent = persona.uni();
>>>>>>> Stashed changes
  }

File: POO/py/Persona.py
class Persona:
    def __init__(self, nombre, edad, carrera, universidad):
        self.nombre = nombre
        self.edad = edad
        self.carrera = carrera
        self.universidad = universidad


    # Método para mostrar la información de la persona
    def mostrar_informacion(self):
        print(f"Nombre: {self.nombre}")
        print(f"Edad: {self.edad}")
        print(f"Carrera: {self.carrera}")
        print(f"Universidad: {self.universidad}")


    # Métodos para modificar los atributos
    def modificar_nombre(self, nuevo_nombre):
        if not nuevo_nombre.strip() or not all(char.isalpha() for char in nuevo_nombre):
         print("Error: El nombre no debe estar vacío ni contener digitos o signos.")
        else:
         self.nombre = nuevo_nombre
        print(f"Nombre ha sido modificado a: {self.nombre}")


    def modificar_edad(self, nueva_edad):
        if not nueva_edad.isdigit() or int(nueva_edad) <= 0 or int(nueva_edad) > 110:
         print("Error: La edad debe ser un número positivo menor a 110 y solo se aceptan digitos.")
        else:
         self.edad = int(nueva_edad)
        print(f"Edad ha sido modificada a: {self.edad}")


    def modificar_carrera(self, nueva_carrera):
        if not nueva_carrera.strip() or not all(char.isalpha() for char in nueva_carrera):
            print("Error: La carrera no debe estar vacio ni contener digitos o signos.")
        else:
            self.carrera = nueva_carrera
            print(f"Carrera ha sido modificado a: {self.carrera}")


    def modificar_universidad(self, nueva_universidad):
        if not nueva_universidad.strip() or not all(char.isalpha() for char in nueva_universidad):
            print("Error. Universidad no debe estar vacio ni contener digitos o signos. ")
        else:
            self.universidad = nueva_universidad
            print(f"Universida ha sido modificado a: {self.universidad}")


    # Métodos para "eliminar" los atributos (asignar None)

    def eliminar_nombre(self):
        self.nombre = None
        print("Nombre ha sido eliminado.")

    def eliminar_edad(self):
        self.edad = None
        print("Edad ha sido eliminada.")

    def eliminar_carrera(self):
        self.carrera = None
        print("Carrera ha sido eliminada.")

    def eliminar_universidad(self):
        self.universidad = None
        print("Universidad ha sido eliminada.")


# Interfaz de usuario en la consola

def menu():
    print("\nMenú de opciones:")
    print("1. Mostrar información")
    print("2. Modificar nombre")
    print("3. Modificar edad")
    print("4. Modificar carrera")
    print("5. Modificar universidad")
    print("6. Eliminar nombre")
    print("7. Eliminar edad")
    print("8. Eliminar carrera")
    print("9. Eliminar universidad")
    print("0. Salir")


# Crear una instancia de la clase Persona

persona = Persona('Miguel', 18, 'Ing. Industrial', 'UCB')

# Ciclo principal de la aplicación

while True:
    menu()
    opcion = input("Seleccione una opción: ")

    if opcion == '1':
        persona.mostrar_informacion()
    elif opcion == '2':
        nuevo_nombre = input("Ingrese el nuevo nombre : ")
        persona.modificar_nombre(nuevo_nombre)
    elif opcion == '3':
        nueva_edad = input("Ingrese la nueva edad : ")
        persona.modificar_edad(nueva_edad)
    elif opcion == '4':
        nueva_carrera = input("Ingrese la nueva carrera : ")
        persona.modificar_carrera(nueva_carrera)
    elif opcion == '5':
        nueva_universidad = input("Ingrese la nueva universidad : ")
        persona.modificar_universidad(nueva_universidad)
    elif opcion == '6':
        persona.eliminar_nombre()
    elif opcion == '7':
        persona.eliminar_edad()
    elif opcion == '8':
        persona.eliminar_carrera()
    elif opcion == '9':
        persona.eliminar_universidad()
    elif opcion == '0':
        print("Saliste del programa")
        break
    else:
        print("Opción no válida. Ingrese una opción del menú")
